crates/biome_js_analyze/src/lint/nursery/no_restricted_imports.rs

```diff
@@ -60,42 +286,705 @@ declare_lint_rule! {
 }
 
 /// Options for the rule `noRestrictedImports`.
-#[derive(Clone, Debug, Default, Deserialize, Deserializable, Eq, PartialEq, Serialize)]
+#[derive(
+    Clone,
+    Debug,
+    Default,
+    Deserialize,
+    biome_deserialize_macros::Deserializable,
+    Eq,
+    PartialEq,
+    Serialize,
+)]
 #[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
 #[serde(rename_all = "camelCase", deny_unknown_fields, default)]
 pub struct RestrictedImportsOptions {
     /// A list of names that should trigger the rule
     #[serde(skip_serializing_if = "FxHashMap::is_empty")]
-    paths: FxHashMap<Box<str>, Box<str>>,
+    paths: FxHashMap<Box<str>, CustomRestrictedImport>,
+}
+
+/// Specifies why a specific import is allowed or disallowed.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+enum ImportRestrictionCause {
+    /// Reason: The import source is forbidden or allowed.
+    ImportSource,
+    /// Reason: A set of forbidden import names has been defined via `importNames`.
+    ImportNames,
+    /// Reason: A set of allowed import names has been defined via `allowImportNames`.
+    AllowImportNames,
+}
+
+/// Specifies whether a specific import is (dis)allowed, and why it is allowed/disallowed.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+struct ImportRestrictionStatus(bool, ImportRestrictionCause);
+
+impl ImportRestrictionStatus {
+    pub fn is_allowed(&self) -> bool {
+        self.0
+    }
+
+    pub fn is_forbidden(&self) -> bool {
+        !self.0
+    }
+
+    pub fn reason(&self) -> ImportRestrictionCause {
+        self.1
+    }
```

Comment:
```suggestion
    pub fn to_reason(&self) -> ImportRestrictionCause {
        self.1
    }
```

Since we return a new instance of `ImportRestrictionCause`, we should use the [`to_` prefix](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv) 

However, I don't think we need to rely on `Copy`, and we could return just a reference